### PR TITLE
Feature/red 506 def file settings

### PR DESCRIPTION
**Pull request by red-tt-bear**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,6 +11,8 @@
     "@material-ui/icons": "^4.11.2",
     "@material-ui/lab": "^4.0.0-alpha.57",
     "@material-ui/pickers": "^3.2.10",
+    "@rederly/rederly-utils": "0.0.2",
```
Comment (PanopticaRising) [inline package.json:14]: Should this be locked at version 0.0.2?

File: src/Courses/TopicSettings/TopicSettingsPage.tsx [hunk 2]
```diff
@@ -202,6 +295,61 @@ export const TopicSettingsPage: React.FC<TopicSettingsPageProps> = ({topic: topi
...
+            <ConfirmationModal
+                headerContent={<h4>Overwrite settings</h4>}
+                cancelText="No"
+                confirmText="Yes"
+                bodyContent={<>
+                    <h6>Do you want to overwrite the following settings:</h6>
+                    <ul>
+                        {Object.entries(translateTopicSettings(topicSettingsOverwriteModalOptions?.examSettings)).map((setting) => (
+                            // <li key={Math.random()}>TOMTOM</li>
+                            <li key={`${setting[0]}-${setting[1]}`}>
```
Comment (PanopticaRising) [inline src/Courses/TopicSettings/TopicSettingsPage.tsx:318]: Do these need to be null-checked, or are they guaranteed cause they come from `entries`?

File: src/Courses/TopicSettings/TopicSettingsPage.tsx [hunk 3]
```diff
@@ -202,6 +295,61 @@ export const TopicSettingsPage: React.FC<TopicSettingsPageProps> = ({topic: topi
...
+                                <strong>{setting[0]}:</strong> {setting[1]}
+                            </li>
+                        ))}
+                    </ul>
+                </>}
+                onConfirm={async ()=>{
+                    try {
+                        if (_.isNil(topicSettingsOverwriteModalOptions)) {
+                            const error = new Error('Options were missing, could not update topic');
+                            logger.error(error);
```
Comment (PanopticaRising) [inline src/Courses/TopicSettings/TopicSettingsPage.tsx:328]: Should the logger be moved to the catch statement to also handle API errors?

File: src/Courses/TopicSettings/TopicSettingsPage.tsx [hunk 4]
```diff
@@ -202,6 +295,61 @@ export const TopicSettingsPage: React.FC<TopicSettingsPageProps> = ({topic: topi
...
+                            throw error;
+                        }
+                        const result = await putTopic({
+                            data: {
+                                topicAssessmentInfo: topicSettingsOverwriteModalOptions.examSettings,
+                                topicTypeId: 2
```
Comment (PanopticaRising) [inline src/Courses/TopicSettings/TopicSettingsPage.tsx:334]: CourseInterfaces should have an enum for this.

**Review by PanopticaRising — DISMISSED** [review]

**Review by PanopticaRising — APPROVED** [review]

File: src/Courses/TopicSettings/TopicSettingsPage.tsx (hunk 2 above)
Comment (red-tt-bear) [reply src/Courses/TopicSettings/TopicSettingsPage.tsx:319]: Added a comment

**Pull request merged**